lib/active_fedora/file.rb

```diff
@@ -139,6 +140,19 @@ def metadata?
       false
     end
 
+    def external_content?
+      external_url.present?
+    end
+
+    def external_url
+      return nil unless mime_type.start_with? "message/external-body"
+      mime_type[/url=['"](.+?)['"]/i, 1]
+    end
+
+    def external_url=(external_file_url)
+      self.mime_type = "message/external-body; access-type=URL; URL=\"#{external_file_url}\""
```

Comment:
If the mime type is `message/external-body` with `access-type=URL` and `URL=anything`, then you can't tell if there is binary content or not (because Fedora will return a redirect instead of the binary).  I guess you could check the SHA1 checksum against the known checksum for an empty binary, but that doesn't seem like a good idea to me.

So I think we should leave the content out and say a resource with external content is valid if all of these are true:
* mime type is `message/external-body`
* the `access-type` parameter is present and the value is `URL`
* the `URL` parameter is present and the value is a valid URL

I'm less sure about the other stuff: `message/external-body` can handle a couple of other URL schemes (ftp, sftp, local-file), and in theory I could imagine someone wanting to use that to point to those.  So I think it would be better to be flexible on that.